torch/_decomp/__init__.py

```diff
@@ -116,6 +116,15 @@ def add_op_to_table(aten_op):
                     and "CompositeImplicitAutograd" not in torch._C._dispatch_dump(name)
                     and not torch._C._dispatch_has_kernel_for_dispatch_key(name, "Meta")
                 ):
+                    if any(a.alias_info is not None and not a.alias_info.is_write for a in op_overload._schema.arguments):
+                        raise RuntimeError(f"""
+Attempting to register a python meta kernel for a view operator: {str(op_overload)}.
+We shouldn't do this, because the output will report as not having aliased storages.
```

Comment:
I would maybe add a note that this is the state of the current implementation but could be fixed in the future.